app/javascript/modules/listingDetails/ListingDetailsUnitAccordions.tsx

```diff
@@ -45,7 +45,18 @@ const getTableData = (units: RailsUnit[]) => {
   })
 }
 
-const sortUnits = (units) => {
+type UnitType = {
+  units: RailsUnit[]
+  availability: number
+  minSqFt: number
+  maxSqFt: number
+}
+
+type PropertyType = {
+  [key: string]: UnitType
+}
+
+const sortUnits = (units: RailsUnit[]): PropertyType => {
```

Comment:
suggestion: We can use the [record utility](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) to save a few lines and remove the need for the `PropertType` type, e.g.
```js
const sortUnits = (units: RailsUnit[]): Record<RailsUnit["Unit_Type"], UnitType> => {
```

Reply:
😍 Amazing!